polaris-patterns/src/components/PolarisPatternsProvider/PolarisPatternsProvider.tsx

```diff
@@ -0,0 +1,57 @@
+import type {PropsWithChildren} from 'react';
+import React, {createContext, useEffect} from 'react';
+import {I18nextProvider, initReactI18next} from 'react-i18next';
+import {createInstance} from 'i18next';
+
+import {DEFAULT_I18N_DETAILS} from '../../configure';
+
+export const I18nDetails = createContext(DEFAULT_I18N_DETAILS);
+
+const i18next = createInstance({
+  fallbackLng: DEFAULT_I18N_DETAILS.locale,
+  interpolation: {
+    escapeValue: false,
+  },
+  resources: {},
+  returnNull: false,
+});
+
+i18next.use(initReactI18next).init();
+
+interface TranslationDictionary {
+  [key: string]: string | TranslationDictionary;
+}
+
+interface I18nDetails {
+  locale: string;
+  country?: string;
```

Comment:
We have a precedent for using `country` to refer to country name and `countryCode` to refer to country code in Shopify Admin ([ref](https://shopify.dev/docs/api/admin-graphql/2022-07/objects/MailingAddress#field-mailingaddress-countrycodev2)), and I wonder if we should keep it that way for consistency. In this case, I suggest using the field name `countryCode` to indicate a two-digit country code.
```suggestion
  countryCode?: string;
```

Reply:
do you think consistency with our apis using `countryCode` is better over consistency with [Intl](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/region) using `region`?

Reply:
That's a good question - `region` might actually be better overall for geopolitical reasons (we also recommend using the label `Country/Region` instead of just `Country` on address forms for the same reasons).